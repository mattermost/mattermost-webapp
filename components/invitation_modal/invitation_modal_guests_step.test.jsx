// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import InvitationModalGuestsStep from './invitation_modal_guests_step.jsx';

describe('components/invitation_modal/InvitationModalGuestsStep', () => {
    test('should match the snapshot', () => {
        const wrapper = shallow(
            <InvitationModalGuestsStep
                teamName='Test Team'
                currentTeamId='123'
                myInvitableChannels={[]}
                searchProfiles={jest.fn()}
                searchChannels={jest.fn()}
                onSubmit={jest.fn()}
                onEdit={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
