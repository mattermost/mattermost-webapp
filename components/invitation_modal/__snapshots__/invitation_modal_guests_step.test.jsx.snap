// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/invitation_modal/InvitationModalGuestsStep should match the snapshot 1`] = `
<div
  className="InvitationModalGuestsStep"
>
  <div
    className="modal-icon"
  >
    <InviteIcon />
  </div>
  <h1>
    <injectIntl(FormattedMarkdownMessage)
      defaultMessage="Invite **Guests**"
      id="invitation_modal.guests.title"
    />
  </h1>
  <div
    className="add-people"
    data-testid="addPeople"
  >
    <h2>
      <FormattedMessage
        defaultMessage="Invite People"
        id="invitation_modal.guests.add_people.title"
        values={Object {}}
      />
    </h2>
    <div
      data-testid="emailPlaceholder"
    >
      <FormattedMessage
        defaultMessage="Add guests or email addresses"
        id="invitation_modal.guests.search-and-add.placeholder"
        values={Object {}}
      >
        <Component />
      </FormattedMessage>
    </div>
    <div
      className="help-text"
    >
      <FormattedMessage
        defaultMessage="Search and add guests or email invite new users."
        id="invitation_modal.guests.add_people.description"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="add-channels"
    data-testid="channelPlaceholder"
  >
    <h2>
      <FormattedMessage
        defaultMessage="Search and Add Channels"
        id="invitation_modal.guests.add_channels.title"
        values={Object {}}
      />
    </h2>
    <div>
      <FormattedMessage
        defaultMessage="Search and add channels"
        id="invitation_modal.guests.add_channels.placeholder"
        values={Object {}}
      >
        <Component />
      </FormattedMessage>
    </div>
    <div
      className="help-text"
    >
      <FormattedMessage
        defaultMessage="Specify the channels the guests have access to."
        id="invitation_modal.guests.add-channels.description"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="custom-message"
    data-testid="customMessage"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Set a custom message"
        id="invitation_modal.guests.custom-message.link"
        values={Object {}}
      />
    </a>
    <div
      className="help-text"
    >
      <FormattedMessage
        defaultMessage="Create a custom message to make your invite more personal."
        id="invitation_modal.guests.custom-message.description"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="invite-guests"
  >
    <button
      className="btn btn-inactive"
      disabled={true}
      id="inviteGuestButton"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Invite Guests"
        id="invitation_modal.guests.invite_button"
        values={Object {}}
      />
    </button>
  </div>
</div>
`;
